hypex/dataset/backends/pandas_backend.py

```diff
@@ -190,11 +191,27 @@ def _get_column_index(
             else self.data.columns.get_indexer(column_name)
         )[0]
 
-    def get_column_type(self, column_name: str) -> str:
-        return str(self.data.dtypes[column_name])
+    def get_column_type(self, column_name: str) -> Union[DefaultRoleTypes, None]:
+        dtype = self.data.dtypes[column_name]
+        if pd.api.types.is_integer_dtype(dtype):
+            return int
+        elif pd.api.types.is_float_dtype(dtype):
+            return float
+        elif pd.api.types.is_string_dtype(dtype) or pd.api.types.is_object_dtype(dtype) or dtype.name == 'category':
+            return str
+        elif pd.api.types.is_bool_dtype(dtype):
+            return bool
+        else:
+            return None
+        
+    def astype(
+        self,
+        dtype: Dict[str, DefaultRoleTypes],
+        errors: Literal["raise", "ignore"] = "raise"
+    ) -> pd.DataFrame:
+        return self.data.astype(dtype=dtype, errors=errors)
 
-    # try try-except if necessary
-    def update_column_type(self, column_name: str, type_name: str):
+    def update_column_type(self, column_name: str, type_name: DefaultRoleTypes):
```

Comment:
type name should be passed as str or type object, not as an object of one of the DefaultRoleTypes